### PR TITLE
fix: login redirect - FS-1653

### PR DESCRIPTION
**Pull request by netbe**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When login in on a fresh install on a federated backend, the user gets redirected to landing screen and has to enter credentials twice to login.

This impacts also automated tests from QA.

### Causes (Optional)

There is a migration for Federated Backend needed as soon as possible once we get the response of apiVersioning and as the user is not authenticated yet, this produces a state transition to unauthenticated flow.

### Solutions

The migration is not needed when switching from different backend domain. We had some logic to do that.
#### How to Test

Manual testing cases:
1. install app
2. open deeplink to dogfood env
3. login

### Notes (Optional)

There is one case which this PR does not handle yet:
1. user was authenticated on a not federated backend A
2. He tries to login to the same backend A but the backend become federated 
-> the user will have to relogin again (happens just once when the migration occurs)

This could be done on a second PR/ticket.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — COMMENTED** [review]
Could we add a unit test for this?

**Review by David-Henner — COMMENTED** [review]
any chance we can cover that with a test ?

**Comment by netbe** [comment]
added of course @David-Henner @johnxnguyen

File: wire-ios-sync-engine/Tests/Source/SessionManager/APIVersionResolverTests.swift [hunk 1]
```diff
@@ -365,6 +372,35 @@ class APIVersionResolverTests: ZMTBaseTest {
         XCTAssertTrue(mockDelegate.didReportFederationHasBeenEnabled)
     }
 
+    func testThatItReportsToDelegate_WhenNotFederatedBackendSwitchedToFederatedBackend() throws {
```
Comment (johnxnguyen) [inline wire-ios-sync-engine/Tests/Source/SessionManager/APIVersionResolverTests.swift:372]:
```suggestion
    func testThatItDoesNotReportToDelegate_WhenNotFederatedBackendSwitchedToFederatedBackend() throws {
```

**Review by MarcinR — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**